controllers/clustergroupupgrade_controller.go

```diff
@@ -898,19 +898,61 @@ func (r *ClusterGroupUpgradeReconciler) getPolicyContent(
 
 			innerObjectDefinitionContent := innerObjectDefinition.(map[string]interface{})
 			// Get the object's metadata.
+			objectDefinitionMetadata := innerObjectDefinitionContent["metadata"]
+			if objectDefinitionMetadata == nil {
+				r.Log.Info(
+					"[getPolicyContent] Policy is missing its spec.policy-templates.objectDefinition.spec.object-templates.metadata",
+					"policyName", managedPolicyName)
+				continue
+			}
+
 			objectDefinitionMetadataContent := innerObjectDefinitionContent["metadata"].(map[string]interface{})
-			// Save the kind, name and namespace.
+			// Save the kind, name and namespace if they exist and if kind is of Subscription type.
+			// If kind is missing, log and skip.
+			_, ok := innerObjectDefinitionContent["kind"]
+			if ok == false {
+				r.Log.Info(
+					"[getPolicyContent] Policy is missing its spec.policy-templates.objectDefinition.spec.object-templates.kind",
+					"policyName", managedPolicyName)
+				continue
+			}
+			// Filter only Subscription templates.
+			if innerObjectDefinitionContent["kind"] != utils.PolicyTypeSubscription {
```

Comment:
you lookup innerObjectDefinitionContent["kind"] twice, in line 912 and 920. 
You could maybe store it in a 'kind' variable in 912 so it can be used in 920.